File: Parallel_And_Distributed_Computing/Dijkstra/Test/Regular.cpp
#include "stdafx.h"
#include "CppUnitTest.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;
using namespace CUDA;

namespace Test
{
	TEST_CLASS(Regular)
	{
	public:

		TEST_METHOD(small_size)
		{
			Graph<double> *graph = new Graph<double>(SMALL_SIZE);
			Dijkstra<double> *dijkstra = new Dijkstra<double>(graph);
			dijkstra->run(0);
			delete graph;
			delete dijkstra;
		}
		TEST_METHOD(middle_size)
		{
			Graph<double> *graph = new Graph<double>(MIDDLE_SIZE);
			Dijkstra<double> *dijkstra = new Dijkstra<double>(graph);
			dijkstra->run(0);
			delete graph;
			delete dijkstra;
		}
		TEST_METHOD(big_size)
		{
			Graph<double> *graph = new Graph<double>(BIG_SIZE);
			Dijkstra<double> *dijkstra = new Dijkstra<double>(graph);
			dijkstra->run(0);
			delete graph;
			delete dijkstra;
		}

	};
}

File: Parallel_And_Distributed_Computing/Dijkstra’sAlgorithm/Test/Regular.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Dijkstras_Algorithm;

namespace Test
{
    [TestClass]
    public class Regular
    {
        [TestMethod]
        public void Small()
        {
            new Dijkstra(new Graph(Tester.SMALL_SIZE).Generate()).Run(from: 0);
        }
        [TestMethod]
        public void Middle()
        {
            new Dijkstra(new Graph(Tester.MIDDLE_SIZE).Generate()).Run(from: 0);
        }
        [TestMethod]
        public void Big()
        {
            new Dijkstra(new Graph(Tester.BIG_SIZE).Generate()).Run(from: 0);
        }
    }
}